rgf/rgf.py

```diff
@@ -516,12 +629,28 @@ def fit(self, X, y):
         np.savetxt(os.path.join(loc_temp, "train.data.x"), X, delimiter=' ', fmt="%s")
         #Store the targets into RGF format
         np.savetxt(os.path.join(loc_temp, "train.data.y"), y, delimiter=' ', fmt="%s")
+        if sample_weight is not None:
+            #Store the weights into RGF format
+            np.savetxt(os.path.join(loc_temp, "train.data.weight"), sample_weight, delimiter=' ', fmt="%s")
+
+        if self.sl2 is None:
+            self.sl2 = self.l2
+
+        if isinstance(self.min_samples_leaf, (float, np.float, np.float16, np.float32, np.float64, np.double)):
```

Comment:
Unfortunately, I don't understand what do you mean. You are right that
> In doc, self.min_samples_leaf is float or int.

When we want to check int we write the following (according to sklearn GitHub):
![image](https://cloud.githubusercontent.com/assets/25141164/26593400/9a041390-456c-11e7-91d8-c54affe0da93.png)

So, in our situation we are checking float and user may use _np.double(0.5)_ or _np.float16(0.5)_. That is why we need so many alternatives, because _isinstance(np.double(0.5), float) == False_